lib/LanguageServerCompletion/Handler/CompletionHandler.php

```diff
@@ -90,16 +91,23 @@ public function completion(TextDocumentItem $textDocument, Position $position, C
             $completionList->isIncomplete = true;
 
             foreach ($suggestions as $suggestion) {
-                /** @var Suggestion $suggestion */
+                $name = $this->suggestionNameFormatter->format($suggestion);
+                $insertText = $suggestion->snippet() ?: $name;
+                $insertTextFormat = $suggestion->snippet() ? InsertTextFormat::SNIPPET : InsertTextFormat::PLAIN_TEXT;
+
                 $completionList->items[] = new CompletionItem(
-                    $this->suggestionNameFormatter->format($suggestion),
+                    $name,
```

Comment:
Should we use `$suggestion->label()` instead ?
It would be more appropriate since it's what `CompletionItem` expect.
And it won't break anything since we will always have either `insertText` or `textEdit` provided, so `label` will never be used by the client to update the buffer.

Reply:
the `label` in Phpactor is probably different from the `name` in some cases (not sure, but can only imagine that I added it for this reason). So I think safer to leave as name

Reply:
I proposed that because in this PR I **always** provide an `insertText`.
For those who (whom ?) don't have the spec in mind this is a summary of what a `CompletionItem` contains:
* `label`: The label of this completion item - mandatory
* `insertText`: A string that should be inserted into a document when selecting this completion
* `insertTextFormat`: The format of the insert text - defaults to `InsertTextFormat.PlainText`
* `textEdit`: An edit which is applied to a document when selecting this completion. When an edit is provided the value of `insertText` is ignored.

So the `label` is mandatory and will only be used as the text to insert in the buffer in last resort.
For the client there is no difference between using `label` or `insertText`, the exact same logic applies.
Given this two things I decided to always provide the `textEdit` using this logic:
* if `Suggestion::snippet()` is not empty use it and set `insertTextFormat` to `InsertTextFormat.Snippet`
* otherwise use `$name` (formatted to ignore the `$` if needed) and set `insertTextFormat` to `InsertTextFormat.PlainText`

Which guarantee us that the client will never use `CompletionItem.label` as text to insert in the buffer.
So we can put whatever we want in it to help identify the item in the completion list: markers like `$`, `()`, `::` or even glyph/icon.

So I think it's pretty safe to do it, because we won't change the behavior, only what is shown in the popup.

Reply:
> For those who (whom ?

`who` :) `to whom it may concern`, `for whom the bell tolls` (maybe imagine it as `whom => for the person that` .. maybe

Reply:
I think it's still better to leave it as `name`, as it's the _clients_ choice if it uses (or can use) `insertText`. `name` is guaranteed to be correct in the "last case".

Reply:
Since the spec says that the client *should* user `insertText` and not *must* use it, I'm fine letting it like that.
But Ii still feel like we private our self from some nice features just to respond to a broken client.
Handling `insertText` is done the same way has handling `label`.